pkg/cmd/cli/cmd/process.go

```diff
@@ -214,14 +212,17 @@ func RunProcess(f *clientcmd.Factory, out io.Writer, cmd *cobra.Command, args []
 		// when user specify the --value
 		if cmd.Flag("value").Changed {
 			values := kcmdutil.GetFlagStringSlice(cmd, "value")
-			injectUserVars(values, cmd, obj)
+			if err := injectUserVars(values, obj); err != nil {
+				return err
```

Comment:
When a user asks for an action, we discussed above that it does not make sense to give the user as much as possible of what they wanted -- if we can detect that they messed something up in their `oc process` invocation, we should let them know and get them to ask again. I wouldn't like it if I did `oc process | oc create -f -`, screwed up some parameter and was not able to re-run the command again as some of the objects had been created already.